src/Perpetuum/Services/MissionEngine/Missions/MissionInProgress.cs

```diff
@@ -92,6 +93,7 @@ public MissionInProgress(Mission mission,IStandingHandler standingHandler,IProdu
             _standingHandler = standingHandler;
             _productionDataAccess = productionDataAccess;
             _zoneManager = zoneManager;
+            _participants = new Lazy<HashSet<Character>>(LoadParticipantsFromDb);
```

Comment:
Please put this inline on the variable declaration as
`private Lazy<HashSet<Character>> _participants = new Lazy<HashSet<Character>>(() => LoadParticipantsFromDb());`

As a response to both your question and mine, it's entirely possible that this call may fail once, then succeed subsequently. In this case, Lazy initialisation is to be avoided because the initialisation call is only made *once*. If it fails, it will not be called again. It's worth checking this by eye before merging this change, because Lazy<T> initialisation ought to be preferred in all other scenarios.

Reply:
Actually, this is a bit weird.  It looks more like they wanted this to be DB-backed:
```cs
        public void AddParticipant(Character doerCharacter)
        {
            if (!spreadInGang) return;
            
            if (!IsParticipant(doerCharacter))
            {
                WriteParticipant(missionGuid, doerCharacter);
                _participants = null;
            }
        }

        public void ForceAddParticipant()
        {
            WriteParticipant(missionGuid, character);
            _participants = null;
        }


        public static void WriteParticipant(Guid missionGuid, Character doerCharacter)
        {
            if (missionGuid == Guid.Empty || doerCharacter == Character.None) return;

            Db.Query().CommandText("missionAddParticipant")
                .SetParameter("@missionGuid", missionGuid)
                .SetParameter("@characterId", doerCharacter.Id)
                .Execute();
        }
```
It is null'ing out the collection completely, and was using the `LazyIntializer` in that Get method which would re-fetch from the DB.
So actually my change is broken X.X  I might rollback to how they had it.

Reply:
Unless there is a way to get the initializer to re-fire, but from reading up on the Lazy<T> docs I wasn't seeing a way with this alone.  In fact the `LazyInitializer` lets you call and use it as you like, which makes more sense knowing that they drop the state held in memory as soon as any participant joins the mission.

Reply:
Madness! Why not just update the hashset in memory? Unless it’s expecting out of sync, external writes?

Revert this for now. This system likely needs refactoring as a whole.